task/backend/scheduler/scheduler.go

```diff
@@ -0,0 +1,78 @@
+package scheduler
+
+import (
+	"context"
+	"time"
+
+	"github.com/influxdata/influxdb"
+)
+
+type Storage interface {
+	// UpdateLatestCompleted will allow us to set a latest completed time for an id.
+	UpdateLatestCompleted(ctx context.Context, id influxdb.ID, lastRun time.Time) error
+
+	// retrieve the latest completed for this schedulable id.
+	// If GetLatestCompleted is empty scheduler will use "now" as its latest completed
```

Comment:
This would prevent you from running historical jobs?

Reply:
So historical jobs can be run strictly inside the executor in this iteration. We could expand the storage options to allow a list of time's we need to attempt to run again? what are you thinking?

Reply:
If GetLatestCompleted is empty, why not return a zero time?
Then the scheduler or whatever can decide what to do based on that, but it knows that it has a zero time.

Reply:
I guess another option would be to return a special error a "NeverCompletedBefore" error that the scheduler (or whoever is asking then knows how to respond.

Reply:
I agree. I think the scheduler could decide how it handles a "not completed" response.